pkg/virt-api/webhooks/mutating-webhook/mutating-webhook.go

```diff
@@ -89,5 +90,5 @@ func ServeMigrationCreate(resp http.ResponseWriter, req *http.Request) {
 }
 
 func ServeClones(resp http.ResponseWriter, req *http.Request) {
-	serve(resp, req, &mutators.CloneCreateMutator{})
+	serve(resp, req, mutators.NewCloneCreateMutator(rand.String))
```

Comment:
@EdDev what do you think about this injection?
Can you think of a better way of doing it?

Reply:
Passing a function seems problematic to me too.
Looking at the implementation of the clone mutator, I see no reason it cannot work with a simple input name.

I can think of two alternatives:
- Create two factories, one that accepts nothing (`NewCloneCreateMutator`) and one that accepts a target-name (`NewCloneCreateMutatorWithTargetName`).
In production the first is called, then it will call the other with a generated target name.
In tests, only the second one is called.
  - This is a general pattern that I use from time to time. Here, it can be slightly adjusted to say something like: `If a target name was provided, do not generate it`. Then it will look like this:
  Production call does not change:
  ```go
  serve(resp, req, &mutators.CloneCreateMutator{})
  ```
  Test call looks like this:
  ```go
  serve(resp, req, &mutators.CloneCreateMutator{TargetName: "123"})
  ```
  It has the disadvantage of exposing that member and the advantage of saving that 2nd factory.
  
- Let `NewCloneCreateMutator` accept a `randomer` interface. Then pass it a concrete that implements the method. This is a more OO solution with proper polymorphism.

I would have chosen the 1st option probably.

BTW, there is another option, but it may be too tricky for a common reader and not Go idiomatic.
It is based on the ability to "override" a method of a base struct. But IMO it is not worth it for this incident.